include/boost/gil/rasterization/ellipse.hpp

```diff
@@ -0,0 +1,181 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+#ifndef BOOST_GIL_RASTERIZATION_ELLIPSE_HPP
+#define BOOST_GIL_RASTERIZATION_ELLIPSE_HPP
+
+#include <array>
+#include <vector>
+#include <iostream>
+
+namespace boost { namespace gil {
+
+/// \defgroup EllipseRasterization
+/// \ingroup Rasterization
+/// \brief Ellipse rasterization algorithms.
+
+/// \ingroup EllipseRasterization
+/// \brief Performs ellipse rasterization using midpoint algorithm.
+struct midpoint_elliptical_rasterizer
+{
+    /// \brief Returns a vector containing co-ordinates of first quadrant points which lie on
+    /// rasterizer trajectory of the ellipse.
+    /// \param semi_axes - Array containing half of lengths of horizontal and vertical axis respectively.
+    std::vector<std::array<std::ptrdiff_t, 2>> obtain_trajectory(
+                                                    const std::array<unsigned int, 2> semi_axes)
+    {
+        // Citation : J. Van Aken, "An Efficient Ellipse-Drawing Algorithm" in IEEE Computer Graphics and Applications, vol. 4, no. 09, pp. 24-35, 1984.
+        // doi: 10.1109/MCG.1984.275994
+        // keywords: {null}
+        // url: https://doi.ieeecomputersociety.org/10.1109/MCG.1984.275994
+        std::vector<std::array<std::ptrdiff_t, 2>> trajectory_points;
+        std::ptrdiff_t x = semi_axes[0], y = 0;
+        
+        // Variables declared on the following line are temporary variables used for improving 
+        // performance since they help in converting all multiplicative operations inside the loop
+        // in to additive/subtractive operations.
+        long long int t1, t2, t3, t4, t5, t6, t7, t8, t9;
+        t1 = semi_axes[0] * semi_axes[0], t2 = 2 * t1, t3 = 2 * t2;
+        t4 = semi_axes[1] * semi_axes[1], t5 = 2 * t4, t6 = 2 * t5;
+        t7 = semi_axes[0] * t5, t8 = 2 * t7, t9 = 0;
```

Comment:
I believe these three are not modified anywhere, so why not 'document' that declaring them as constants
```cpp
long long int const t1 = ...;
long long int const t4 = ...;
long long int const t7 = ...;
```